tests/callbacks/test_device_stats_monitor.py

```diff
@@ -85,6 +102,128 @@ def log_metrics(self, metrics: Dict[str, float], step: Optional[int] = None) ->
     trainer.fit(model)
 
 
+@RunIf(min_gpus=1)
+def test_device_stats_gpu_from_nvidia_and_cpu(tmpdir):
+    """Test GPU stats + CPU stats are logged using a logger."""
+    model = BoringModel()
+    device_stats = DeviceStatsMonitor()
+
+    class DebugLogger(CSVLogger):
+        @rank_zero_only
+        def log_metrics(self, metrics: Dict[str, float], step: Optional[int] = None) -> None:
+            # Just need to check one of the GPU metrics to make sure
+            # the GPU metrics are logged
+            fields = [
+                "active.all.allocated",
+            ] + CPU_METRIC_KEYS
+            for f in fields:
+                assert any(f in h for h in metrics)
+
+    trainer = Trainer(
+        default_root_dir=tmpdir,
+        max_epochs=2,
+        limit_train_batches=2,
+        log_every_n_steps=1,
+        accelerator="gpu",
+        devices=1,
+        callbacks=[device_stats],
+        logger=DebugLogger(tmpdir),
+        enable_checkpointing=False,
+        enable_progress_bar=False,
+    )
+
+    trainer.fit(model)
+
+
+@RunIf(min_gpus=1)
+def test_device_stats_gpu_from_nvidia_no_cpu(tmpdir):
+    """Test only GPU stat stats are logged using a logger."""
```

Comment:
```suggestion
    """Test only GPU stat stats are logged using a logger."""
```
also update the docstrings here if you merge them